sdk/v2/api_client.go

```diff
@@ -0,0 +1,43 @@
+package sdk
+
+import (
+	"github.com/brigadecore/brigade/sdk/v2/authx"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/sdk/v2/system"
+)
+
+// APIClient is the general interface for the Brigade API. It does little more
+// than expose functions for obtaining more specialized clients for different
+// areas of concern, like User management or Project management.
+type APIClient interface {
+	Authx() authx.APIClient
+	Core() core.APIClient
+	System() system.APIClient
+}
+
+type apiClient struct {
+	authxClient  authx.APIClient
+	coreClient   core.APIClient
+	systemClient system.APIClient
+}
+
+// NewAPIClient returns a Brigade client.
+func NewAPIClient(apiAddress, apiToken string, allowInsecure bool) APIClient {
```

Comment:
I also don't like unwieldy argument lists, but a benefit to the functions signatures as they stand are that you cannot instantiate a client without passing in the _required_ info `apiAddress` and `apiToken`. If those were fields in a config struct of some sort, the possibility would exist that they are accidentally left unset. I like the approach you're suggesting for _optional_ fields.

So... I think we split the difference on this maybe and say that the `allowInsecure` field moves into a new `APIClientOptions` struct, because it's not invalid to casually leave that unset-- the default behavior would be what you'd expect: that insecure connections are _not_ ignored.

Function signature would then look like this:

```
func NewAPIClient(apiAddress, apiToken string, opts restmachinery.APIClientOptions) APIClient
```

And I like this much better than what I originally had, too, because it allows for future expansion of the options without breaking the SDK.

Great suggestion!

Reply:
Nice, I think that's a great compromise!

Reply:
Reviewed 2aaffbf3d69a46e6b2c4a323804a33787d4a7ef9 - LGTM, thank you!